tests/test-team/template-mgmt-component-tests/email/template-mgmt-submit-email-page.component.ts

```diff
@@ -0,0 +1,148 @@
+import { test, expect } from '@playwright/test';
+import SessionStorageHelper from '../../helpers/session-storage-helper';
+import { TemplateMgmtSubmitEmailPage } from '../../pages/email/template-mgmt-submit-email-page';
+import { SessionFactory } from '../../helpers/session-factory';
+import { TemplateStorageHelper } from '../../helpers/template-storage-helper';
+import {
+  assertGoBackLink,
+  assertLoginLink,
+  assertNotifyBannerLink,
+  assertSkipToMainContent,
+} from '../template-mgmt-common.steps';
+
+const templateIds = new Set<string>();
+
+const getAndStoreTemplateId = (url: string) => {
+  const id = String(url.split('/').pop());
+  templateIds.add(id);
+  return id;
+};
+
+const fields = {
+  emailTemplateName: 'test-template-email',
+  emailTemplateSubjectLine: 'test-template-subject-line',
+  emailTemplateMessage: 'test-template-message',
+};
+
+const sessions = {
+  empty: SessionFactory.createEmailSession('empty-email-submit-session'),
+  submit: {
+    ...SessionFactory.createEmailSession('submit-email-submit-session'),
+    ...fields,
+  },
+  submitAndReturn: {
+    ...SessionFactory.createEmailSession('submit-and-return-email-session'),
+    ...fields,
+  },
+  valid: {
+    ...SessionFactory.createEmailSession('valid-email-submit-session'),
+    ...fields,
+  },
+};
+
+test.describe('Submit Email message template Page', () => {
```

Comment:
Maybe don't need this top level test.describe as they are already grouped by being in the same file.

Reply:
Totally your call on this, I'm happy either way.

I quite like the report saying "Submit Email message template page > Page functionality > when user does X, then Y happens".

Reply:
Just a thought on the nested describe. 

We're calling `.beforeAll` and `afterAll` in the top level describe. So it has some use. 

If you like, I can remove the inner test.describe; `Page functionality` and `Error handling`